### DISCUSSION TITLE
Vercel preview deployment domain error with Firebase auth

### DISCUSSION THREAD

**Discussion by pulkitvyas08** [vercel/next.js#37636] [Help]
I use Firebase for auth of my NextJS application that is deployed on Vercel. Now for every preview link that is generated by pull requests, I need to add the new domain to the auth section of the Firebase console otherwise this is the error I get:

``` 
Info: The current domain is not authorized for OAuth operations. This will prevent signInWithPopup, signInWithRedirect, linkWithPopup and linkWithRedirect from working. Add your domain (horse-s-mouth-website.vercel.app) to the OAuth redirect domains list in the Firebase console -> Auth section -> Sign in method tab.
```

And auth actions cannot be performed. Is there any way to automatically add the preview domains from Vercel to the Firebase auth domain list?

**Comment by robertseghedi** [comment]
In case anyone stumbles across this, the js sdk console message Info: The current domain is not authorized for OAuth operations. This will prevent signInWithPopup, signInWithRedirect, linkWithPopup and linkWithRedirect from working. was a red herring and provided the wrong information.

The reason why I was getting this message was because I didn't have the firebase app domain itself (*.firebaseapp.com) included in the Accept requests from these HTTP referrers (web sites) list in the Google API credentials for the app key (https://console.developers.google.com/apis/credentials/key). This must be a fairly recent requirement as accepting requests from just my actual site domain was enough beforehand.

Got the answer [from here](https://github.com/firebase/firebase-js-sdk/issues/998).

**Reply by pulkitvyas08** [reply]
Hi thanks for the reply but my problem is a bit different. The thing is you can add custom domains to the authorized domains list on firebase but how do you automate that for something like Vercel preview links where you will have a new preview link for every PR or re-deployment?

**Comment by khalid-hisham** [comment]
@pulkitvyas08 Did you find a solution to this problem?

**Reply by pulkitvyas08** [reply]
Not yet. I switched to Firebase hosting for all my projects that require OAuth through firebase since it automatically adds all deployment URLs to their authorized domains list

**Reply by khalid-hisham** [reply]
I added vercel.app to the trusted auth domains of my firebase auth and it worked for all the different subdomains of the preview channels afterwards

**Reply by pulkitvyas08** [reply]
Oh okay, that's amazing to hear, thanks!

**Reply by sipec** [reply]
adding vercel.app fixed it for me too. thanks!

**Reply by nenadV91** [reply]
Worked for me also, thanks a lot 🙏

**Reply by drummerjolev** [reply]
This doesn't work for me. Has anyone tried something else?

When using Firebase hosting, the preview environments are added to the Authorized Domains list including their preview subdomain. So I'm thinking this may be the reason my Vercel preview environments aren't working.

**Reply by eddyhdzg-solarx** [reply]
I'm using Vercel preview deployments and it is not working for me.

**Reply by abdirahmanhassann** [reply]
> I added vercel.app to the trusted auth domains of my firebase auth and it worked for all the different subdomains of the preview channels afterwards

this solved my problem.thank you!

**Reply by Susk4** [reply]
Hi, I tried adding vercel.app as well, but I get 'A valid domain name is required (e.g. 'myapp.com'). ' Can anyone help ?

**Reply by SuRiRONHeaRt** [reply]
Dudes, just go to firebase authentication page and click settings. Under domains section, there is option called "Authorized Domains". Add your Vercel App link to the Authorised domains by clicking on Add Domain Button. Works like a charm.  Thanks for the heads-up Abdil

**Reply by epark10** [reply]
This does not address the issue when each new Vercel preview is a different link, unless someone has been able to address this?

**Reply by guilhermeutzig** [reply]
I can confirm that having `vercel.com` added into Firebase made it work across all different Vercel URLs for me.

![image](https://github.com/vercel/next.js/assets/36337530/aa48e16e-88a7-462c-a54f-9de4d32dbe46)

**Comment by kasvith** [comment]
Hi, we came across same problem and developed following github action package to solve it in our CI environment.

https://github.com/wavezync/firebase-authorized-domains

**Reply by naga-k** [reply]
Thanks for acutally addressing the issue. Seems like the others are just explaing the error or suggesting to add vercel.app to the authorized domains.

**Reply by dimi-colossal** [reply]
Please don't add `vercel.app` to your firebase. Adding `vercel.app` to your authorised domains means anyone can use your public firebase credentials to access your firebase project given that they are sending a request from some subdomain of the `vercel.app` domain. Anyone deploying a preview branch on vercel can use your firebase project.
